robottelo/entities.py

```diff
@@ -669,19 +669,79 @@ class Meta(object):
         server_modes = ('sat')
 
 
-class ContentViewPuppetModule(orm.Entity):
+class ContentViewPuppetModule(
+        orm.Entity, orm.EntityReadMixin, orm.EntityDeleteMixin,
+        orm.EntityCreateMixin):
     """A representation of a Content View Puppet Module entity."""
+    author = entity_fields.StringField()
     content_view = entity_fields.OneToOneField('ContentView', required=True)
     name = entity_fields.StringField()
-    author = entity_fields.StringField()
-    uuid = entity_fields.StringField()
+    puppet_module = entity_fields.OneToOneField('PuppetModule')
 
     class Meta(object):
         """Non-field information about this entity."""
-        api_path = ('katello/api/v2/content_views/:content_view_id/'
-                    'content_view_puppet_modules')
         server_modes = ('sat')
 
+    def __init__(self, **kwargs):
+        """Ensure ``content_view`` is passed in and set ``self.Meta.api_path``.
+
+        :raises TypeError: If ``content_view`` is not passed in.
+
+        """
+        if 'content_view' not in kwargs:
+            raise TypeError(
+                'The "content_view" parameter must be provided.'
+            )
+        super(ContentViewPuppetModule, self).__init__(**kwargs)
+        self.Meta.api_path = '{0}/content_view_puppet_modules'.format(
+            self.content_view.path('self')  # pylint:disable=no-member
+        )
+
+    def read(
+            self,
```

Comment:
Why not use the same formatting like https://github.com/SatelliteQE/robottelo/pull/1980/files#diff-51f3c66d9ea6bdd9cd953b251332cd7aR672?